transmogrifier/sources/datacite.py

```diff
@@ -188,59 +201,62 @@ def create_from_datacite_xml(
                 kind=identifier_xml["identifierType"],
             ),
         ]
-        alternate_identifiers = xml.metadata.find_all("alternateIdentifier")
-        for alternate_identifier in alternate_identifiers:
-            i = timdex.Identifier(
-                value=alternate_identifier.string,
+        for alternate_identifier in [
+            i for i in xml.metadata.find_all("alternateIdentifier") if i.string
+        ]:
+            kwargs["identifiers"].append(
+                timdex.Identifier(
+                    value=alternate_identifier.string,
+                    kind=alternate_identifier.get("alternateIdentifierType"),
+                )
             )
-            if "alternateIdentifierType" in alternate_identifier.attrs:
-                i.kind = alternate_identifier.attrs["alternateIdentifierType"]
-            kwargs["identifiers"].append(i)
 
         related_identifiers = xml.metadata.find_all("relatedIdentifier")
         for related_identifier in [
             i
             for i in related_identifiers
-            if "relationType" in i.attrs and i.attrs["relationType"] == "IsIdenticalTo"
+            if i.get("relationType") == "IsIdenticalTo" and i.string
         ]:
             i = timdex.Identifier(
                 value=cls.generate_related_item_identifier_url(related_identifier),
-                kind=related_identifier.attrs["relationType"],
+                kind=related_identifier.get("relationType"),
             )
             kwargs["identifiers"].append(i)
 
         # language
         language = xml.metadata.find("language")
-        if language:
+        if language and language.string:
             kwargs["languages"] = [language.string]
 
         # locations
-        locations = xml.metadata.find_all("geoLocationPlace")
-        for location in locations:
+        for location in [
+            gl for gl in xml.metadata.find_all("geoLocationPlace") if gl.string
+        ]:
             kwargs.setdefault("locations", []).append(
                 timdex.Location(value=location.string)
             )
 
         # notes
         descriptions = xml.metadata.find_all("description")
-        for description in descriptions:
+        for description in [d for d in descriptions if d.string]:
```

Comment:
The `descriptions` variable is set outside the list comprehension here but inside it in most of the other cases. Was that intentional?

Reply:
The `descriptions` variable also gets used for `summary` so that's why I kept it outside of the list comprehension